### DISCUSSION TITLE
amqp_client failures on OTP 26

### DISCUSSION THREAD

**Discussion by jared-mackey** [rabbitmq/rabbitmq-server#8611] [Other] [answered]
### Describe the bug

Hi, I am using `amqp_client` and `amqp` hex packages (via Broadway) for my Elixir project and I was upgrading to Elixir 1.15 and OTP 26 and noticed that connections are failing to connect to my RabbitMQ servers.

Elixir Logs

```
module=BroadwayRabbitMQ.Producer [error] Cannot connect to RabbitMQ broker: {:auth_failure, ~c"Disconnected"}
```

RabbitMQ Server logs

```
[error] <0.818.0> {handshake_error,starting,26721,
[error] <0.818.0>                  {amqp_error,frame_error,
[error] <0.818.0>                              "type 99, all octets = <<>>: {frame_too_large,1852269919,4088}",
[error] <0.818.0>                              none}}
```

### Reproduction steps

1. Use an AMQP elixir/erlang client to connect to RabbitMQ on OTP 26.
2. Open connection to RabbitMQ server. (Mine is running in docker, `rabbitmq:3-management`) 
3. Connections fail with logs on Elixir side and on RabbitMQ server.

### Expected behavior

Connections should not close and app should function normally.

### Additional context

Versions

Old
```
elixir 1.15.0-otp-25
erlang 25.3
```

New that throws the errors:
```
elixir 1.15.0-otp-26
erlang 26.0.1
```

My RabbitMQ server is not using these changes and is just using w/e is in docker at the time I pulled. I can get that exact version if it's helpful but I think it might be client side since it hasn't been an issue before upgrading my client.

**Comment by mkuratczyk** [comment, marked as answer]
Use RabbitMQ 3.12:  https://www.rabbitmq.com/which-erlang.html

The latest 3.11 patch release should also work if can't use 3.12 yet.

**Comment by asyncmind0** [comment]
same error with `amqp_client:3.12.2` and RabbitMQ `3.12.2` on Erlang `25.3.2.5` [jit]

replicated using the official docker image ` rabbitmq:3.12.2`

```rabbitmq  | 2023-08-16 06:30:41.814571+00:00 [info] <0.898.0> accepting AMQP connection <0.898.0> (172.20.0.1:37408 -> 172.20.0.6:5672)
rabbitmq  | 2023-08-16 06:30:44.885041+00:00 [error] <0.898.0> closing AMQP connection <0.898.0> (172.20.0.1:37408 -> 172.20.0.6:5672):
rabbitmq  | 2023-08-16 06:30:44.885041+00:00 [error] <0.898.0> {handshake_error,starting,26721,
rabbitmq  | 2023-08-16 06:30:44.885041+00:00 [error] <0.898.0>                  {amqp_error,frame_error,
rabbitmq  | 2023-08-16 06:30:44.885041+00:00 [error] <0.898.0>                              "type 99, all octets = <<>>: {frame_too_large,1852269919,4088}",
rabbitmq  | 2023-08-16 06:30:44.885041+00:00 [error] <0.898.0>                              none}}
```

**Reply by mkuratczyk** [reply]
Can you provide an executable way of reproducing this issue? Thanks

**Reply by asyncmind0** [reply]
This is the code that is failing

```  
  Params =
    #amqp_params_network{
      username = Username,
      password = Password,
      host = Host,
      port = Port
    },
  logger:debug("Connecting to rabbit worker queue... params: ~p", [Params]),
  {ok, Connection} = amqp_connection:start(Params),

```

debug:

```
Connecting to rabbit worker queue... params: {amqp_params_network,
                                                 <<"rabbit">>,<<"password">>,
                                                 <<"/">>,"zenbook",5672,2047,
                                                 0,10,60000,none,
                                                 [fun amqp_auth_mechanisms:plain/3,
                                                  fun amqp_auth_mechanisms:amqplain/3],
                                                 [],[]}

```

**Reply by michaelklishin** [reply]
Please paste more complete log files.

Such frame parsing errors often indicate that a client without TLS enabled tries to connect to a TLS-enabled port, or vice versa.

You haven't provided the [effective node configuration](https://rabbitmq.com/configure.html#verify-configuration-effective-configuration) nor the actual connection code. We do not guess in this community.

There are also dedicated troubleshooting guides: [Troubleshooting Connectivity](https://rabbitmq.com/troubleshooting-networking.html) and [Troubleshooting TLS](https://rabbitmq.com/troubleshooting-ssl.html).

**Reply by michaelklishin** [reply]
And [taking a traffic capture](https://rabbitmq.com/amqp-wireshark.html) is never a bad idea when you see a parser error in the logs.

**Comment by michaelklishin** [comment]
Team RabbitMQ has discovered a [very serious bug](https://github.com/erlang/otp/issues/7566) (for those who implement binary protocols anyway) in the Erlang compiler that affects 26 on aarch64.

Using Erlang 25.3 may be a safer bet for now.
